tools/prepare_msd.py

```diff
@@ -0,0 +1,165 @@
+    },  # 3d zyx
+    10: {
+        "Task10_Colon.tar":
+        "https://bj.bcebos.com/v1/ai-studio-online/netdisk/062aa5a52cc44597a87f56c5ef1371c7acb52f73a2c946be9fea347dedec5058?responseContentDisposition=attachment%3B%20filename%3DTask10_Colon.tar&authorization=bce-auth-v1%2F0ef6765c1e494918bc0d4c3ca3e5c6d1%2F2022-01-21T18%3A42%3A03Z%2F-1%2F%2F106546582e748224f0833e100fc74d1bf3ff7fe4f4370d43bb487b10c3f5deae"
+    },  # 3d zyx
+}
+
+
+class Prep_msd(Prep):
+
+    def __init__(self, task_id):
+        self.dataset_name = list(tasks[task_id].keys())[0].split(".")[0]
+        super().__init__(
+            dataset_fdr=f"{self.dataset_name}",
+            urls=tasks[task_id],
+            image_fdr=f"{self.dataset_name}/{self.dataset_name}/imagesTr",
+            label_fdr=f"{self.dataset_name}/{self.dataset_name}/labelsTr",
+            phase_fdr="phase0",
+        )
+
+    def convert_path(self):
+        """convert medical imaging file to numpy array in the right directory"""
+
+        print("Start convert images to numpy array, please wait patiently")
+        self.load_save(
+            self.image_dir,
+            save_path=self.image_path,
+            preprocess=[
+                HUNorm,
+                functools.partial(resample, new_shape=[128, 128, 128],
+                                  order=1),
+            ],
+        )
+        print("start convert labels to numpy array, please wait patiently")
+
+        self.load_save(
+            self.label_dir,
+            self.label_path,
+            preprocess=[
+                functools.partial(resample, new_shape=[128, 128, 128],
+                                  order=0),
+            ],
+            tag="label",
+        )
+
+    def generate_txt(self, train_split=15):
+        """generate the train_list.txt and val_list.txt"""
+
+        txtname = [
+            osp.join(self.phase_path, "train_list.txt"),
+            osp.join(self.phase_path, "val_list.txt"),
+        ]
+
+        image_files = os.listdir(self.image_path)
+        label_files = [
+            name.replace("_org_covid-19-pneumonia-",
+                         "_").replace("-dcm", "").replace("_org_", "_")
+            for name in image_files
+        ]
+
+        self.split_files_txt(txtname[0], image_files, label_files)
+        self.split_files_txt(txtname[1], image_files, label_files)
+
+
+# 3d: 2 3 4 6 7 8 9 10
+
+if __name__ == "__main__":
+    assert (
+        len(sys.argv) == 2
+    ), "Please specify msd task id. \n usage: python tools/prepare_msd.py task_id"
+    prep = Prep_msd(task_id=int(sys.argv[1]))
+    prep.uncompress_file(num_zipfiles=1)
+    prep.convert_path()
+    prep.generate_txt()
```

Comment:
Did this script do well in the test?

Reply:
Download, uncompress and load_saving 3d dataset should work in this version. 

Maybe you guys can test out the download part again. Downloading from bos is awfully slow here, that's the main reason I wrote skip already downloaded file.

Preprocessing 4d sequence is in a later pr

Reply:
ok. We will test it when this commit is separated from others.